app/src/main/java/com/github/sdp/ratemyepfl/viewmodel/AddReviewViewModel.kt

```diff
@@ -71,13 +79,13 @@ class AddReviewViewModel @Inject constructor(
 
         // only connected users may add reviews
         if (!connectedUser.isLoggedIn()) {
-            return null
-        } else if (comment == null || comment == "") {
-            return null
-        } else if (title == null || title == "") {
-            return null
+            throw DisconnectedUserException()
```

Comment:
Good point. Even if Kotlin allows easy management of nullable, it is a good idea to throw an Exception. I think that you can change the return type to non-nullable.

Reply:
I did not see that tests were failing. From what I read, it seems that the merge with the main failed (or is incomplete).

Reply:
yeah Ill try to figure out where it comes from